src/main.ts

```diff
@@ -0,0 +1,146 @@
+            eventName = core.getInput('event_id');
+        }
+
+        switch (eventName) {
+            case EventType.PUSH:
+                parsedSchema = schemaOnpush.validateSync(eventPayloadText);
+                account = parsedSchema.pusher.name;
+                return {
+                    message: `**Commit to GitHub** by ${account}`,
+                    url: parsedSchema.compare,
+                    details: `Comment: ${parsedSchema.head_commit.message}`
+                };
+            case EventType.PULL_REQUEST:
+                parsedSchema = schemaOnPullRequest.validateSync(eventPayloadText);
+                account = parsedSchema.pull_request.user.login;
+                return {
+                    message: `**PR submitted to Github** by ${account} with title: ${parsedSchema.pull_request.title}`,
+                    url: parsedSchema.pull_request.html_url,
+                    details: `Pr for merging ref ${parsedSchema.pull_request.head.ref} to base branch ${parsedSchema.pull_request.base.ref}`
+                };
+            case EventType.ISSUE:
+                parsedSchema = schemaOnIssue.validateSync(eventPayloadText);
+                account = parsedSchema.sender.login;
+                return {
+                    message: `**New/updated GitHub issue**: ${parsedSchema.issue.title}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            case EventType.ISSUE_COMMENT:
+                parsedSchema = schemaOnIssueComment.validateSync(eventPayloadText);
+                return {
+                    message: `**A Github issue comment was posted** by ${account} with content ${parsedSchema.comment.body}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            default:
+                return {
+                    message: eventName ? eventName.replace(/_/g, '.') : 'A GitHub Actions event has occurred',
+                    url: `https://github.com/${repository}/actions`,
+                    details: core.getInput('details')
+                };
+        }
+    } catch (error) {
+        let errorDetails;
+        if (error instanceof ValidationError) {
+            errorDetails = error.errors;
+        } else {
+            errorDetails = error.message;
+        }
+        return core.setFailed(`ERROR : ${errorDetails}`);
+    }
+};
+
+
+async function notifyTeams(notificationMessage: NotificationMessage) {
+    if (!(core.getInput('webhook_url'))) {
+        core.setFailed('Error : Webhook URL configuration is missing. Aborting');
+        process.exit(1);
+    }
+    const matches = core.getInput('webhook_url').match(/^https?:\/\/([^/?#]+)(.*)/i);
+    const hostnameMatch = matches && matches[1];
```

Comment:
how come you break down the `'webhook_url'` into `hostname` and `path`? can't use it directly as-is?

Reply:
I did not have to, I opted for the https://nodejs.org/api/https.html#https_https_request_options_callback using the separate host/port/path options instead of the whole URL passed as the options object.
In any case, using node-fetch this has been removed.